init.go

```diff
@@ -1,89 +1,133 @@
 package main
 
 import (
-	"bytes"
 	"flag"
 	"fmt"
-	"io"
 	"log"
 	"os"
-	"os/exec"
-	"path/filepath"
-	"sync"
-	"syscall"
+
+	"github.com/storageos/init/k8s"
+	"github.com/storageos/init/k8s/inspector"
+	"github.com/storageos/init/script"
+	"github.com/storageos/init/script/runner"
+)
+
+const (
+	daemonSetNameEnvVar       = "DAEMONSET_NAME"
+	daemonSetNamespaceEnvVar  = "DAEMONSET_NAMESPACE"
+	nodeImageEnvVar           = "NODE_IMAGE"
+	defaultDaemonSetName      = "storageos-daemonset"
+	defaultDaemonSetNamespace = "kube-system"
+	defaultContainerName      = "storageos"
 )
 
 func main() {
 	scriptsDir := flag.String("scripts", "", "absolute path of the scripts directory")
+	dsName := flag.String("dsName", "", "name of the StorageOS DaemonSet")
+	dsNamespace := flag.String("dsNamespace", "", "namespace of the StorageOS DaemonSet")
+	nodeImage := flag.String("nodeImage", "", "container image of StorageOS Node, use when running out of k8s")
 
 	flag.Parse()
 
+	// StorageOS node container image.
+	var storageosImage string
+
+	// Abort if no scripts directory is provided.
 	if *scriptsDir == "" {
 		log.Println("no scripts directory specified, pass scripts dir with -scripts flag.")
 		os.Exit(1)
 	}
 
-	allScripts, err := getAllScripts(*scriptsDir)
+	// Attempt to get storageos node image.
+
+	if *nodeImage == "" {
+		// This is in k8s environment. Get the node image from StorageOS k8s
+		// resources.
+		kubeclient, err := k8s.NewK8SClient()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// Create a k8s inspector.
+		inspect := inspector.NewInspect(kubeclient)
+
+		img, err := getImageFromK8S(inspect, *dsName, *dsNamespace)
+		if err != nil {
+			log.Fatal(err)
+		}
+		storageosImage = img
+	} else {
+		storageosImage = *nodeImage
+	}
+
+	// Abort if storageos node image is still unknown.
+	if storageosImage == "" {
+		log.Println("unknown storageos node image, pass node image with -nodeImage flag.")
+		os.Exit(1)
+	}
```

Comment:
No specific reason. Since there's a flag for passing the scripts dir path, I decided to add options for other parameters too. Flags are used in the tests for now. But we can change that if we prefer to use just env vars.